### PR TITLE
[14.0][MIG] product_attribute_value_archive

### PR DESCRIPTION
**Pull request by mmequignon**

### CODE REVIEW

**Comment by sebalix** [comment]
Port of https://github.com/OCA/product-variant/pull/228

File: product_attribute_value_archive/tests/test_archive_attribute_value.py [hunk 1]
```diff
@@ -75,18 +74,22 @@ def setUpClassCommon(cls):
...
+        for product in products:
+            vals.append(
+                {
+                    "product_id": product.id,
+                    "product_uom_qty": 1,
+                    "order_id": cls.order.id,
+                }
+            )
+        cls.env["sale.order.line"].create(vals)
+        return cls.order
```
Comment (sebalix) [inline product_attribute_value_archive/tests/test_archive_attribute_value.py:92]: We should avoid impure functions, here you update a variable (`cls.order`) that is declared outside of the method, while the name of the method and its parameters suggests you are creating order lines only.
Why did you change the previous implementation?

If there is a good reason to change the implementation, I would write:
```python
def _create_sale_order_lines(cls, order, products):
    ...
    return order
```
And pass explicitly the order on each call.

**Review by sebalix — CHANGES_REQUESTED** [review]

File: product_attribute_value_archive/tests/test_archive_attribute_value.py (hunk 1 above)
Comment (mmequignon) [reply product_attribute_value_archive/tests/test_archive_attribute_value.py:92]: The issue is that, sale_stock is installed by another module and introduces a required field on sale_order.
However, we just have the constraint on the database, but the fields doesn't exists in the odoo context.
Then, odoo doesn't set the default value on this fields, and creating sale order fails.
I've found this workaround : adding so lines to on an existing sale order.

File: product_attribute_value_archive/tests/test_archive_attribute_value.py [hunk 2]
```diff
@@ -75,18 +74,22 @@ def setUpClassCommon(cls):
             ]
         )
         cls.red_pen = pen_template_value.ptav_product_variant_ids
+        cls.order = cls.env.ref("sale.sale_order_1")
```
Comment (sebalix) [inline product_attribute_value_archive/tests/test_archive_attribute_value.py:77]: Then this line deserves a comment:
```python
# NOTE: use an existing order and avoid the creation a new one
# to avoid NOT NULL constraint introduced by other addons
```

**Review by sebalix — CHANGES_REQUESTED** [review]
